spark-indexer/src/main/java/org/uniprot/store/spark/indexer/taxonomy/mapper/TaxonomyStatisticsAggregationMapper.java

```diff
@@ -4,20 +4,25 @@
 import org.uniprot.core.taxonomy.TaxonomyStatistics;
 import org.uniprot.core.taxonomy.impl.TaxonomyStatisticsBuilder;
 import org.uniprot.store.spark.indexer.common.util.SparkUtils;
+import org.uniprot.store.spark.indexer.taxonomy.mapper.model.TaxonomyStatisticsWrapper;
 
 public class TaxonomyStatisticsAggregationMapper
-        implements Function2<TaxonomyStatistics, TaxonomyStatistics, TaxonomyStatistics> {
+        implements Function2<
+                TaxonomyStatisticsWrapper, TaxonomyStatisticsWrapper, TaxonomyStatisticsWrapper> {
 
     private static final long serialVersionUID = -2085055845000585882L;
 
     @Override
-    public TaxonomyStatistics call(TaxonomyStatistics stat1, TaxonomyStatistics stat2)
+    public TaxonomyStatisticsWrapper call(
+            TaxonomyStatisticsWrapper wrapper1, TaxonomyStatisticsWrapper wrapper2)
             throws Exception {
-        TaxonomyStatistics mergedStat = null;
-        if (SparkUtils.isThereAnyNullEntry(stat1, stat2)) {
-            mergedStat = SparkUtils.getNotNullEntry(stat1, stat2);
+        TaxonomyStatisticsWrapper mergedWrapper = null;
+        if (SparkUtils.isThereAnyNullEntry(wrapper1, wrapper2)) {
+            mergedWrapper = SparkUtils.getNotNullEntry(wrapper1, wrapper2);
         } else {
-            mergedStat =
+            TaxonomyStatistics stat1 = wrapper1.getStatistics();
```

Comment:
can we name stat1/2 differently to reflect the statistics they're for?

Reply:
Hi Edd, This Statistic Aggregation Mapper is used to aggregate, protein stats with another protein stats, and also is used to aggregate proteome stats with another proteome stats. This is why it is difficult to get a good name for it.

Reply:
I will merge it and we can try to figure out a better name for it afterwards.